README.md

```diff
@@ -1 +1,142 @@
-# broadcast
\ No newline at end of file
+# Broadcast
+> Abstract module to send broadcast messages.
+
+[![Build Status](https://travis-ci.com/dxos/broadcast.svg?branch=master)](https://travis-ci.com/dxos/broadcast)
+[![Coverage Status](https://coveralls.io/repos/github/dxos/broadcast/badge.svg?branch=master)](https://coveralls.io/github/dxos/broadcast?branch=master)
+![npm (scoped)](https://img.shields.io/npm/v/@dxos/broadcast)
+[![Greenkeeper badge](https://badges.greenkeeper.io/dxos/broadcast.svg)](https://greenkeeper.io/)
+[![js-semistandard-style](https://img.shields.io/badge/code%20style-semistandard-brightgreen.svg?style=flat-square)](https://github.com/standard/semistandard)
+[![standard-readme compliant](https://img.shields.io/badge/readme%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
+
+Allows a node to originate a message that will be received at least once, within a
+reasonably short time, on all nodes that are reachable from the origin node. Messages are
+propagated via the `middleware` specified. Broadcast storms are
+avoided by means of a [flooding](https://en.wikipedia.org/wiki/Flooding_(computer_networking)) routing scheme.
+
+Broadcast messages follows the schema:
+
+```proto
+message Packet {
+  bytes seqno = 1;
+  bytes origin = 2;
+  bytes from = 3;
+  bytes data = 4;
+}
+```
+
+- `seqno`: By default is a random 32-bit but could be used to provide an alternative sorted sequence number.
+- `origin`: Represents the author's ID of the message. To identify a message (`msgId`) in the network you should check for the: `seqno + origin`.
+- `from`: Represents the current sender's ID of the message.
+- `data`: Represents an opaque blob of data, it can contain any data that the publisher wants
+it to defined by higher layers (e.g. a presence information message).
+
+Nodes send any message originating locally to all current peers. Upon receiving a message, a
+node delivers it locally to any listeners, and forward the message on to its current
+peers, excluding the peer from which it was received.
+
+Nodes maintain a record of the messages they have received and originated
+recently, by `msgId(seqno + from)`. This is used to avoid sending the same message to the same peer
+more than once. These records expire after some time to limit memory consumption by: `maxAge` and `maxSize`.
+
+<p align="center">
+  <img src="https://user-images.githubusercontent.com/819446/66934639-2bb67980-f011-11e9-9c27-739b5ee5fd5c.gif" alt="graph">
+</p>
+
+## Install
+
+```
+$ npm install @dxos/broadcast
+```
+
+## Usage
+
+```javascript
+import { Broadcast } from '@wirelineio/broadcast';
+
+const middleware = {
```

Comment:
Please talk with Max and me about READMEs. I don't like them. Too easy to break.  I prefer better test and example files.

Reply:
@richburdon I am in favor of APIs in READMEs. Not a complete guide on how to use it, but for such a small modules / apis like these I don't see any downside in having these types of READMEs. They also are very useful for offline reference (IMO)

Reply:
I agree to keep having simple `APIs` and `usages` in the readme. Good reading about the standard readme that we are using: https://github.com/RichardLitt/standard-readme